### PR TITLE
Ftr/parts in go to menu

### PR DESCRIPTION
**Pull request by feuerbart**
## Description, Context and related Issue
* add parts to goto-menu in source view.
* add preference to change order from partwise to movementwise

Refs #13 
Refs #271 

## How Has This Been Tested?

run in local Edirom-Online, tested with open-faust sources A2 (without parts) and A3 (with parts).

## Types of changes
- New feature (non-breaking change which adds functionality)
- Improvement

## Overview
- I have updated the inline documentation accordingly.
- I have performed a self-review of my code, according to the [style guide](https://github.com/Edirom/Edirom-Online/blob/develop/STYLE-GUIDE.md)
- I have read the [CONTRIBUTING](https://github.com/Edirom/Edirom-Online/blob/develop/CONTRIBUTING.md) document.

### CODE REVIEW

File: add/data/xql/getMovements.xql [hunk 1]
```diff
@@ -27,7 +27,14 @@ let $movements as array(*)* :=
         return
             map {
                 'id': $movement/string(@xml:id),
-                'name': $movement/string(@label)
+                'name': $movement/string(@label),
+	            'parts': for $part in $movement//mei:part
```
Comment (peterstadler) [inline add/data/xql/getMovements.xql:31]: two comments here:
1. cosmetic: please adjust the indentation 
2. I think parts should always be an array to facilitate processing on the JS-frontend. While sequences of maps will properly be casted to JSON arrays when serializing as JSON, a single part would be serialized as a JSON object, not an array. I suggest to wrap the `for $part … ` into an `array { …}`.

**Review by peterstadler — CHANGES_REQUESTED** [review]

**Comment by feuerbart** [comment]
fixed the indentation and added the array-wrap.
tested again with open-faust A2, A3 and A4.

File: add/data/xql/getMovements.xql (hunk 1 above)
Comment (peterstadler) [reply add/data/xql/getMovements.xql]: Thanks for the prompt fix!

**Review by peterstadler — COMMENTED** [review]
The code looks good to me but I haven't had the chance to check the display and the promised new functionality. Someone with parts data should check this in a running Edirom.

**Comment by daniel-jettka** [comment]
@riedde Could you have a look at this one?

File: add/data/xql/getMovements.xql [hunk 2]
```diff
@@ -27,7 +27,16 @@ let $movements as array(*)* :=
         return
             map {
                 'id': $movement/string(@xml:id),
-                'name': $movement/string(@label)
+                'name': $movement/string(@label),
+                'parts': array {
+                    for $part in $movement//mei:part
+                    return
+                        map {
+                            'id': $part/string(@xml:id),
```
Comment (riedde) [inline add/data/xql/getMovements.xql:35]: We don't have a Fallback, if there are no parts. In https://klarinettenquintett.weber-gesamtausgabe.de/index.html (ED1) there are no part IDs and the menu is shown but nor working. Maybe hide the menue in that case? The alternative would be to generate IDs for that case.
Using the Baumann-Digital Data, it works (the parts there have IDs).

**Comment by peterstadler** [comment]
btw, is this also related to #101?

**Comment by feuerbart** [comment]
yes, #101 adresses the same feature.

**Comment by feuerbart** [comment]
added a fix to the issue @riedde mentioned: parts will only be shown if they have ids so it wont break if they don't have.

**Review by riedde — APPROVED** [review]
Thank you @feuerbart !

**Review by peterstadler — APPROVED** [review]

**Pull request merged**